controllers/metal3.io/host_state_machine.go

```diff
@@ -542,17 +542,23 @@ func (hsm *hostStateMachine) handleDeprovisioning(info *reconcileInfo) actionRes
 			return actionComplete{}
 		}
 
+		skipToPowerOff := func() actionResult {
+			hsm.NextState = metal3api.StatePoweringOffBeforeDelete
+			info.postSaveCallbacks = append(info.postSaveCallbacks, deleteWithoutDeprov.Inc)
+			return actionComplete{}
+		}
+
 		switch r := actResult.(type) {
 		case actionComplete:
-			hsm.NextState = metal3api.StateDeleting
+			hsm.NextState = metal3api.StatePoweringOffBeforeDelete
 			hsm.Host.Status.ErrorCount = 0
 		case actionFailed:
 			// If the provisioner gives up deprovisioning and
 			// deletion has been requested, continue to delete.
 			if hsm.Host.Status.ErrorCount > 3 {
 				info.log.Info("Giving up on host clean up after 3 attempts. The host may still be operational " +
 					"and cause issues in your clusters. You should clean it up manually now.")
-				return skipToDelete()
+				return skipToPowerOff()
```

Comment:
Given that `skipToPowerOff` and `skipToDelete` are only used one time each, I would suggest either inlining them or making the next state a parameter and having a single function.

Reply:
Done; inlined the functions, and also changed the prometheus counter to be `deleteWithoutPowerOff` in the case of `skipToDelete`.